packages/shorebird_cli/test/src/commands/build/build_apk_command_test.dart

```diff
@@ -33,43 +36,39 @@ void main() {
     late ProcessResult processResult;
     late BuildApkCommand command;
     late ShorebirdFlutterValidator flutterValidator;
-
-    String? processExecutable;
-    List<String>? processArguments;
+    late ShorebirdProcess shorebirdProcess;
 
     setUp(() {
       argResults = _MockArgResults();
       httpClient = _MockHttpClient();
       auth = _MockAuth();
       logger = _MockLogger();
+      shorebirdProcess = _MockShorebirdProcess();
       processResult = _MockProcessResult();
       flutterValidator = _MockShorebirdFlutterValidator();
-      processExecutable = null;
-      processArguments = null;
       command = BuildApkCommand(
         auth: auth,
         logger: logger,
-        runProcess: (
-          executable,
-          arguments, {
-          bool runInShell = false,
-          Map<String, String>? environment,
-          String? workingDirectory,
-          bool useVendedFlutter = true,
-        }) async {
-          processExecutable = executable;
-          processArguments = arguments;
-          return processResult;
-        },
         validators: [flutterValidator],
-      )..testArgResults = argResults;
-
+      )
+        ..testArgResults = argResults
+        ..testProcess = shorebirdProcess
+        ..testEngineConfig = const EngineConfig.empty();
+      registerFallbackValue(shorebirdProcess);
```

Comment:
minor but this could be extracted into `setUpAll` since it only needs to run once.

Reply:
Just the registerFallbackValue?  I guess.  It woudl be more lines of code. :)

Reply:
haha yeah it's not a huge deal either way just left a comment for context.